src/mapper/pkg/pod_webhook/certs.go

```diff
@@ -0,0 +1,90 @@
+package pod_webhook
+
+import (
+	"context"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func copyCA(ctx context.Context, client client.Client, secretName, secretNamespace, targetNamespace string) error {
+	exists, err := checkCAExists(ctx, client, secretName, targetNamespace)
+
+	if err != nil {
+		err = fmt.Errorf("unable to check if configmap %s exists in namespace %s: %w", secretName, targetNamespace, err)
+		logrus.Error(err)
+		return err
+	}
+
+	if exists {
+		logrus.Debug("CA configmap already exists - no action required")
+		return nil
+	}
+
+	var secret corev1.Secret
+	err = client.Get(ctx, types.NamespacedName{
+		Namespace: secretNamespace,
+		Name:      secretName,
+	}, &secret)
+
+	if err != nil {
+		err = fmt.Errorf("unable to get certificate authority secret %s in namespace %s: %w", secretName, secretNamespace, err)
+		logrus.Error(err)
+		return err
+	}
+
+	certificate := secret.Data["ca.crt"]
+
+	err = writeCA(ctx, client, secretName, targetNamespace, certificate)
+
+	if err != nil {
+		err = fmt.Errorf("unable to write certificate authority to configmap %s in namespace %s: %w", secretName, targetNamespace, err)
+		logrus.Error(err)
+		return err
+	}
+
+	logrus.Infof("copied CA to configmap %s in namespace %s", secretName, targetNamespace)
+	return nil
+}
+
+func checkCAExists(ctx context.Context, client client.Client, configMapName, configMapNamespace string) (bool, error) {
+	var configMap corev1.ConfigMap
+	err := client.Get(ctx, types.NamespacedName{
+		Namespace: configMapNamespace,
+		Name:      configMapName,
+	}, &configMap)
+
+	if err == nil {
```

Comment:
Checking for err == nil is usually frowned upon, as the pattern is to check for err != nil, and it's error prone to write code that has the inverted logic. Switch this section to check for if err != nil